packages/back-end/src/models/TeamModel.ts

```diff
@@ -0,0 +1,105 @@
+import mongoose from "mongoose";
+import { omit } from "lodash";
+import uniqid from "uniqid";
+import { TeamInterface } from "../../types/team";
+
+const teamSchema = new mongoose.Schema({
+  id: {
+    type: String,
+    unique: true,
+  },
+  name: String,
+  organization: String,
+  dateCreated: Date,
+  dateUpdated: Date,
+  createdBy: String,
+  description: String,
+  role: String,
+  limitAccessByEnvironment: Boolean,
+  environments: [String],
+  projectRoles: [
+    {
+      _id: false,
+      project: String,
+      role: String,
+      limitAccessByEnvironment: Boolean,
+      environments: [String],
+    },
+  ],
+});
+
+type TeamDocument = mongoose.Document & TeamInterface;
+
+const TeamModel = mongoose.model<TeamInterface>("Team", teamSchema);
+
+type UpdateTeamProps = Omit<
+  TeamInterface,
+  "dateCreated" | "dateUpdated" | "id" | "organization" | "createdBy"
+>;
+
+/**
+ * Convert the Mongo document to a TeamInterface, omitting Mongo default fields __v, _id
+ * @param doc
+ */
+const toInterface = (doc: TeamDocument): TeamInterface => {
+  return omit(doc.toJSON<TeamDocument>(), ["__v", "_id"]);
+};
+
+export async function createTeam(
+  data: Partial<TeamInterface>
+): Promise<TeamInterface> {
+  const teamDoc = await TeamModel.create({
+    ...data,
+    id: uniqid("team_"),
+    dateCreated: new Date(),
+    dateUpdated: new Date(),
+  });
+  return toInterface(teamDoc);
+}
+
+export async function findTeamById(
+  id: string,
+  orgId: string
+): Promise<TeamInterface | null> {
+  const teamDoc = await TeamModel.findOne({ id, organization: orgId });
+  return teamDoc ? toInterface(teamDoc) : null;
+}
+
+export async function getTeamsForOrganization(
+  orgId: string
+): Promise<TeamInterface[]> {
+  const teamDocs = await TeamModel.find({
+    organization: orgId,
+  });
+  return teamDocs.map((team) => toInterface(team));
+}
+
+export async function updateTeamMetadata(
+  id: string,
+  orgId: string,
+  update: UpdateTeamProps
+): Promise<UpdateTeamProps> {
+  const changes = {
+    ...omit(update, "members"),
```

Comment:
Is the `...omit` needed? It doesn't look like the endpoint that calls `updateTeamMetadata` takes in the `members`.

Reply:
Nope! Removing this